solidjs-tailwind/src/auth/preventUnauthorised.js

```diff
@@ -0,0 +1,13 @@
+import { useLocation, useNavigate } from '@solidjs/router';
+
+const preventUnauthorised = (authStore, redirectPath) => () => {
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  if (!authStore.isAuthenticated) {
+    sessionStorage.setItem('auth_return_path', location.pathname);
```

Comment:
WHat's happening here? Why do we need to do this?

Reply:
This way we know where we came from on Signin.jsx:22 . It helps us to return the user to their origin.